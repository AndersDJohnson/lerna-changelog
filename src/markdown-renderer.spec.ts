import MarkdownRenderer from "./markdown-renderer";
import {CommitInfo} from "./interfaces";

describe("MarkdownRenderer", () => {
  describe("groupByCategory", () => {
    it("group commits by category", () => {
      const renderer = new MarkdownRenderer({
        baseIssueUrl: 'http://foo.bar',
        categories: [
          ":rocket: New Feature",
          ":boom: Breaking Change",
          ":bug: Bug Fix",
          ":nail_care: Enhancement",
          ":memo: Documentation",
          ":house: Maintenance",
        ],
      });
      const testCommits = [
        { commitSHA: "a0000005", categories: [] },
        { commitSHA: "a0000004", categories: [":bug: Bug Fix"] },
        { commitSHA: "a0000003", categories: [":rocket: New Feature"] },
        { commitSHA: "a0000002", categories: [] },
        { commitSHA: "a0000001", categories: [":bug: Bug Fix"] },
      ];
      const commitsByCategory = renderer.groupByCategory(testCommits as CommitInfo[]);

      expect(commitsByCategory).toMatchSnapshot();
    });
  });
});
