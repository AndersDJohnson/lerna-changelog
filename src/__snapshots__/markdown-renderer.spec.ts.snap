// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownRenderer groupByCategory group commits by category 1`] = `
Array [
  Object {
    "commits": Array [
      Object {
        "categories": Array [
          ":rocket: New Feature",
        ],
        "commitSHA": "a0000003",
      },
    ],
    "name": ":rocket: New Feature",
  },
  Object {
    "commits": Array [],
    "name": ":boom: Breaking Change",
  },
  Object {
    "commits": Array [
      Object {
        "categories": Array [
          ":bug: Bug Fix",
        ],
        "commitSHA": "a0000004",
      },
      Object {
        "categories": Array [
          ":bug: Bug Fix",
        ],
        "commitSHA": "a0000001",
      },
    ],
    "name": ":bug: Bug Fix",
  },
  Object {
    "commits": Array [],
    "name": ":nail_care: Enhancement",
  },
  Object {
    "commits": Array [],
    "name": ":memo: Documentation",
  },
  Object {
    "commits": Array [],
    "name": ":house: Maintenance",
  },
]
`;
